### PR TITLE
feat(dashboards) : duplicate a dashboard 

### PR DESCRIPTION
**Pull request by Med1233**
## Description

Dashboard: duplicate

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Dashboards/api/useCreateDashboard.ts [hunk 1]
```diff
@@ -23,16 +28,34 @@ type UseCreateDashboard<
   'mutate' | 'mutateAsync'
 >;
 
-const useCreateDashboard = (): UseCreateDashboard => {
+interface Labels {
+  labelFailure: string;
+  labelSucces: string;
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/api/useCreateDashboard.ts:33]:
```suggestion
  labelSuccess: string;
```

File: centreon/www/front_src/src/Dashboards/api/useCreateDashboard.ts [hunk 2]
```diff
@@ -23,16 +28,34 @@ type UseCreateDashboard<
...
   } = useMutationQuery<Omit<Dashboard, 'globalRefreshInterval'>, unknown>({
     getEndpoint: () => dashboardsEndpoint,
+    httpCodesBypassErrorSnackbar: [400, 500],
     method: Method.POST,
-    mutationKey: [resource.dashboards, 'create']
+    mutationKey: [resource.dashboards, 'create'],
+    ...(labels?.labelFailure
+      ? { onError: () => showErrorMessage(labels?.labelFailure) }
+      : {}),
+    ...(labels?.labelSucces
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/api/useCreateDashboard.ts:56]:
```suggestion
    ...(labels?.labelSuccess
```

File: centreon/www/front_src/src/Dashboards/api/useCreateDashboard.ts [hunk 3]
```diff
@@ -23,16 +28,34 @@ type UseCreateDashboard<
...
+      ? { onSuccess: () => showSuccessMessage(labels?.labelSucces) }
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/api/useCreateDashboard.ts:57]:
```suggestion
      ? { onSuccess: () => showSuccessMessage(labels?.labelSuccess) }
```

File: centreon/www/front_src/src/Dashboards/hooks/useDashboardDuplicate.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+type UseDashboardForm = (name) => void;
+
+const useDashboardDuplicate = (): UseDashboardForm => {
+  const { t } = useTranslation();
+
+  const dashboardToDuplicate = useAtomValue(dashboardToDuplicateAtom);
+
+  const labels = {
+    labelFailure: t(labelFailedToDuplicateDashboard),
+    labelSucces: t(labelDashboardDuplicated)
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/hooks/useDashboardDuplicate.tsx:25]:
```suggestion
    labelSuccess: t(labelDashboardDuplicated)
```

File: centreon/packages/ui/src/components/DataTable/DataTable.cypress.spec.tsx [hunk 5]
```diff
@@ -27,14 +23,18 @@ const initializeDataTableGrid = ({
...
             key={title}
-            labelsDelete={{
-              cancel: 'Cancel',
-              confirm: {
-                label: 'Delete'
-              }
+            labels={{
+              labelDelete: 'Delete',
+              labelDuplicate: 'Duplicate',
+              labelEditProperties: 'Edit Properties',
```
Comment (bdauria) [inline centreon/packages/ui/src/components/DataTable/DataTable.cypress.spec.tsx:29]: can we use the actual label variables ? so it's less error prone

File: centreon/packages/ui/src/components/DataTable/DataTable.cypress.spec.tsx [hunk 6]
```diff
@@ -109,32 +108,33 @@ describe('DataTable: Grid', () => {
...
-    cy.findAllByLabelText('edit access rights').should('have.length', 5);
-    cy.findAllByLabelText('edit').should('have.length', 5);
+    cy.findAllByLabelText('Share with contacts').should('have.length', 5);
+    cy.findAllByLabelText('More actions').should('have.length', 5);
 
     cy.makeSnapshot();
   });
 
-  it('displays items with delete action', () => {
+  it('displays actions when more actions button was clicked', () => {
```
Comment (bdauria) [inline centreon/packages/ui/src/components/DataTable/DataTable.cypress.spec.tsx:121]:
```suggestion
  it('displays actions when more actions button is clicked', () => {
```

File: centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx [hunk 7]
```diff
@@ -66,33 +86,63 @@ const DataTableItem = forwardRef(
...
+                <MoreIcon />
+              </IconButton>
+              <Menu
+                anchorEl={moreActionsOpen}
+                open={Boolean(moreActionsOpen)}
+                onClose={closeMoreActions}
+              >
+                <ActionsList
+                  actions={[
+                    ...(onEdit
```
Comment (bdauria) [inline centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx:114]: as the logic becomes slightly more complex, it'd be a good idea to start extracting it out in my opinion

Comment (Med1233) [reply centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx:114]: Yes, you are right. However, this component was initially designed to serve the dashboards exclusively. Nonetheless, I have created a refactoring ticket with the purpose of integrating this component as part of the Listing component, and then adding a prop variant = 'List' | 'Grid'

**Review by bdauria — APPROVED** [review]

File: centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx [hunk 8]
```diff
@@ -66,33 +86,64 @@ const DataTableItem = forwardRef(
...
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx:114]: @bdauria was right. IMO, the best option is to make the more action generic. Why not defining a prop that allow the develop to a list of icons, labels and actions?

**Review by smau-centreon — DISMISSED** [review]

File: centreon/www/front_src/src/Dashboards/components/DashboardLibrary/DashboardCardActions/useDashboardCardActions.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import { Dashboard } from '../../../api/models';
+import { useDashboardConfig } from '../DashboardConfig/useDashboardConfig';
+
+interface Props {
+  dashboard: Dashboard;
+}
+
+interface useDashboardCardActionsState {
+  closeMoreActions: () => void;
+  moreActionsOpen;
```
Comment (bdauria) [inline centreon/www/front_src/src/Dashboards/components/DashboardLibrary/DashboardCardActions/useDashboardCardActions.ts:19]: boolean?

File: centreon/tests/e2e/features/Dashboards/03-dashboard-properties-edition/index.ts [hunk 10]
```diff
@@ -112,8 +114,11 @@ Given(
     cy.visit('/centreon/home/dashboards');
     cy.contains(dashboardToEdit.name)
       .parent()
-      .find('button[aria-label="edit"]')
+      .find('button[aria-label="More actions"]')
       .click();
+
+    cy.get('[aria-label="Edit properties"]').click();
```
Comment (paul-oureib) [inline centreon/tests/e2e/features/Dashboards/03-dashboard-properties-edition/index.ts:120]:
```suggestion
    cy.getByLabel({ label: 'Edit properties', tag: 'button' }).click();
```
should be applied on every instance of this change

**Review by MedWassCh — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**